### PR TITLE
Add context logging, make sure any actions the agent takes are logged at info level

### PR DESCRIPTION
**Pull request by adjackura**
Added context logging library and moved almost all logging calls to it.
Any labels added get added as labels to any log call that uses the ctx or child context
My plan is to move this up to the core logging library once it supports all the features we want.

### CODE REVIEW

File: agentendpoint/patch_windows.go [hunk 1]
```diff
@@ -128,24 +128,25 @@ func (r *patchTask) wuaUpdates(ctx context.Context) error {
 }
 
 func (r *patchTask) runUpdates(ctx context.Context) error {
-	// Don't use retry function as wuaUpdates handles it's own retries.
-	if err := r.wuaUpdates(ctx); err != nil {
-		return err
-	}
-
+	// Install GooGet updates first as this will allow us to update the agent prior to any potential WUA bugs/errors.
```
Comment (ryanwe) [inline agentendpoint/patch_windows.go:131]: Should this be a separate PR?

File: main.go [hunk 2]
```diff
@@ -110,9 +97,14 @@ func run(ctx context.Context) {
 		fmt.Printf("Error initializing logger: %v", err)
 		os.Exit(1)
 	}
-	packages.DebugLogger = log.New(&logWriter{}, "", 0)
+	ctx = clog.WithLabels(ctx, map[string]string{"instance_name": config.Name()})
```
Comment (ryanwe) [inline main.go:100]: Isn't this already included in the log event since it comes from the instance?

File: ospatch/apt_upgrade.go [hunk 3]
```diff
@@ -89,12 +91,12 @@ func RunAptGetUpgrade(opts ...AptGetUpgradeOption) error {
 		pkgNames = append(pkgNames, pkg.Name)
 	}
 	logger.Infof("Updating %d packages.", len(pkgNames))
-	logger.Debugf("Packages to be installed: %s", fPkgs)
+	logger.Infof("Packages to be installed: %s", fPkgs)
```
Comment (ryanwe) [inline ospatch/apt_upgrade.go:94]: It may also be clearer to combine this with the above log line since they're both `Info` now. Maybe` "Updating %d packages: %s", len(pkgNames), fPkgs `. Or you could go further and combine it with the "dryrun" message too. So for dryruns, log `Dryrun, not updating %d packages: %s", len(pkgNames), fPkgs `.

And fyi that people may code against this. So even if we add structured logging in the future, we may need to keep this log message around.

File: ospatch/yum_update.go [hunk 4]
```diff
@@ -106,12 +108,12 @@ func RunYumUpdate(opts ...YumUpdateOption) error {
 		pkgNames = append(pkgNames, pkg.Name)
 	}
 	logger.Infof("Updating %d packages.", len(pkgNames))
-	logger.Debugf("Packages to be installed: %s", fPkgs)
+	logger.Infof("Packages to be installed: %s", fPkgs)
```
Comment (ryanwe) [inline ospatch/yum_update.go:111]: Same comment as above.

File: ospatch/zypper_patch.go [hunk 5]
```diff
@@ -146,22 +148,22 @@ func RunZypperPatch(opts ...ZypperPatchOption) error {
 		logger.Infof("No patches to install.")
 	} else {
 		logger.Infof("Installing %d patches.", len(fPatches))
-		logger.Debugf("Patches to be installed: %s", fPatches)
+		logger.Infof("Patches to be installed: %s", fPatches)
```
Comment (ryanwe) [inline ospatch/zypper_patch.go:151]: Same comment as above.

File: ospatch/zypper_patch.go [hunk 6]
```diff
@@ -146,22 +148,22 @@ func RunZypperPatch(opts ...ZypperPatchOption) error {
...
 	}
 
 	if len(fpkgs) == 0 {
 		logger.Infof("No non-patch packages to update.")
 	} else {
 		logger.Infof("Updating %d packages.", len(fpkgs))
-		logger.Debugf("Packages to be installed: %s", fpkgs)
+		logger.Infof("Packages to be installed: %s", fpkgs)
```
Comment (ryanwe) [inline ospatch/zypper_patch.go:158]: here too. This should also allow you to remove the `if len() == 0` check.

File: retryutil/retry.go [hunk 7]
```diff
@@ -94,7 +95,7 @@ func RetryAPICall(maxRetryTime time.Duration, name string, f func() error) error
 			return err
 		}
 
-		logger.Errorf("Error calling %s, attempt %d, retrying in %s: %v", name, i, ns, err)
+		clog.Errorf(ctx, "Error calling %s, attempt %d, retrying in %s: %v", name, i, ns, err)
```
Comment (ryanwe) [inline retryutil/retry.go:98]: Should this be a Warning since we're still retrying?

File: main.go (hunk 2 above)
Comment (adjackura) [reply main.go:100]: Thats just instance id unfortunately.

File: ospatch/apt_upgrade.go (hunk 3 above)
Comment (adjackura) [reply ospatch/apt_upgrade.go:94]: Ack on the structured logging. What I think i may end up doing is keeping the message field as the human readable form, and then adding more structure around that. So still one log entry just with both this message and also more structured fields.

File: clog/clog.go [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+	labels map[string]string
+	ctx    context.Context
+}
+
+func (l *log) log(msg string, sev logger.Severity) {
+	logger.Log(logger.LogEntry{Message: msg, Severity: sev, CallDepth: 4, Labels: l.labels})
+}
+
+// Debugf simulates logger.Debugf and adds context labels.
+func Debugf(ctx context.Context, format string, args ...interface{}) {
```
[the remark this answers is not in the document]
Comment (adjackura) [reply clog/clog.go:43]: We can't make it invisible to customers, if you set the log level that level will be used, we can add more debug levels, so level 1 is less verbose, 2 is more, 3 is even more and so on. If we want to add this I can do it in another PR.
